cmake/external/googletest.cmake

```diff
@@ -0,0 +1,55 @@
+# Copyright 2017 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(ExternalProject)
+ExternalProject_Add(
+  googletest_external
+  PREFIX googletest
+  GIT_REPOSITORY "https://github.com/google/googletest.git"
+  GIT_TAG "release-1.8.0"
+  GIT_SHALLOW ON
```

Comment:
This line causes my cmake to attempt to clone googletest.git into ON/GIT_SHALLOW. (The 'ON' directory doesn't exist, so it fails.)

Should this work? (I tested on my gubuntu desktop. So not exactly an apple system.)

UPDATE: Works fine on my macbook, as well as my (home) ubuntu 17.04 machine, so perhaps this is fine. Maybe the minimum cmake version is wrong?

Reply:
Looks like `GIT_SHALLOW` was added considerably later than 2.8 (some patch level of 3.6). I added this mostly because a full clone of grpc can take close to a minute and we don't really need the history.

I'd like to support Ubuntu back to Trusty without using a whole lot of version-checking portability gunk so I've changed this to use a URL instead which achieves the desired fast download without the versioning headache.